runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java

```diff
@@ -514,8 +524,24 @@ public DataflowPipelineJob run(Pipeline pipeline) {
     LOG.info("Executing pipeline on the Dataflow Service, which will have billing implications "
         + "related to Google Compute Engine usage and other Google Cloud Services.");
 
-    List<DataflowPackage> packages = options.getStager().stageFiles();
+    List<DataflowPackage> packages = options.getStager().stageDefaultFiles();
+
+    RunnerApi.Pipeline protoPipeline = PipelineTranslation.toProto(pipeline);
+    File serializedProtoPipeline;
```

Comment:
Also, this change assumes we have write access to a temporary directory on the users system. We currently require read only access to be able to submit a Dataflow pipeline, do you think we should be able to use FileSystems to write to GCS location directly and not use the stager class directly?

Reply:
It is a good point. I'll see how much I have to reimplement to get that going.

Reply:
Done